lib/asyn-thread.c

```diff
@@ -302,6 +302,14 @@ query_complete(DWORD err, DWORD bytes, LPWSAOVERLAPPED overlapped)
   struct Curl_addrinfo *ca;
   struct Curl_addrinfo *cafirst = NULL;
   struct Curl_addrinfo *calast = NULL;
+#ifndef CURL_DISABLE_SOCKETPAIR
+#ifdef USE_EVENTFD
+  const void *buf;
+  static const uint64_t val = 1;
```

Comment:
Why static? Won't having this static risk causing problems in multi-threaded situations?

Reply:
it's read-only, so it should be fine.

Reply:
When there are multiple callers, sharing the variable should be a good thing in general, right? Given that these functions can get called a lot.

Reply:
Is it? If it just exists on the stack it might be much cheaper than having it static - and either way, I doubt you could even measure a performance difference if you tried. I actually don't know the exact winner here but I know that using `static` is a red flag in code in general and it makes the code stand out less if you avoid it.

Reply:
Convinced, we will go with the stack memory.